aea/mail/base.py

```diff
@@ -715,20 +726,27 @@ async def _send(self, envelope: Envelope) -> None:
         :raises ValueError: if the connection id provided is not valid.
         :raises AEAConnectionError: if the connection id provided is not valid.
         """
+        connection_id = None  # type: Optional[PublicId]
         envelope_context = envelope.context
-        connection_id = (
-            envelope_context.connection_id if envelope_context is not None else None
-        )
+        # first, try to route by context
+        if envelope_context is not None:
+            connection_id = envelope_context.connection_id
+
+        # second, try to route by default routing
+        if connection_id is None and envelope.protocol_id in self.default_routing:
+            connection_id = self.default_routing[envelope.protocol_id]
+            logger.debug("Using default routing: {}".format(connection_id))
```

Comment:
Idea is, that optionally the user can define a default mapping, so they don't have to set context on each envelope. For instance, `{PublicId.from_str("fetchai/oef_search:0.1.0"):  PublicId.from_str("fetchai/soef:0.1.0")}` could be set when I want to run the agent with the p2p_p2plib and soef connections. Then all outgoing traffic will go to `p2p_libp2p` unless it is a search envelope, which will go to `soef`.

Reply:
OK, sounds good.
An idea for the future. Could a protocol be associated with multiple connections? Meaning that the message of a protocol must be sent to each connection. Basically, the routing relationship would become a many-to-many association. The same should be possible via envelope context. The use cases would be quite rare though.

Reply:
Yes it definitely could. But I think, as you say, it is likely a rare case, and already supported if you use the `EnvelopeContext` for routing (which takes precedence !). I think this is a good first iteration as it covers the current issues we have without making it a massive feature.